lib/testutils/src/lib.rs

```diff
@@ -414,13 +414,25 @@ impl<'settings, 'repo> CommitGraphBuilder<'settings, 'repo> {
         write_random_commit(self.mut_repo, self.settings)
     }
 
-    pub fn commit_with_parents(&mut self, parents: &[&Commit]) -> Commit {
+    pub fn commit_builder_with_parents(&mut self, parents: &[&Commit]) -> CommitBuilder {
         let parent_ids = parents
             .iter()
             .map(|commit| commit.id().clone())
             .collect_vec();
-        create_random_commit(self.mut_repo, self.settings)
-            .set_parents(parent_ids)
+        create_random_commit(self.mut_repo, self.settings).set_parents(parent_ids)
+    }
+
+    pub fn commit_with_parents(&mut self, parents: &[&Commit]) -> Commit {
+        self.commit_builder_with_parents(parents).write().unwrap()
+    }
+
+    pub fn commit_with_parents_and_tree(
+        &mut self,
+        parents: &[&Commit],
+        tree: &MergedTree,
+    ) -> Commit {
+        self.commit_builder_with_parents(parents)
+            .set_tree_id(tree.id())
```

Comment:
To show that these are worth defining centrally, can you update existing callers of `commit_with_parents()` to use these new functions where appropriate?

Reply:
There would be no users of such a function since commit_with_parents returns a commit, which means they can't modify the tree. I assume this means they shouldn't be defined centrally, so I've just defined them in the test case.